designs/per_command_help_text.md

```diff
@@ -0,0 +1,439 @@
+
+  **This command does not require/requires access to AWS credentials.**
+  
+More detailed explanation.
+
+Workflows:
+    $ sam command -> **sam sub-command**
+
+Acronyms:
+    IAM : Identity and Access Management.
+    ARN: Amazon Resource Name.
+    SNS: Simple Notification Service.
+    ECR: Elastic Container Registy
+    KMS: Key Management Service.
+
+Options:
+
+    **Required Options:**
+        ...
+    AWS Credential Options: (Ordering changes on a per command basis)
+        ....
+    Configuration Options:
+        ....
+    Infrastructure Options:
+        ....
+    Additional Options:
+        ....
+    Verbosity Options:
+        ....
+    
+```
+
+The sync help text could be changed to look like below. Only the options are showcased here.
+
+```bash
+Usage: sam sync [OPTIONS]
+
+  NEW! Sync an AWS SAM Project to AWS.
+
+  Examples:
+  $ sam sync --code --stack-name {stack} --resource-id \
+  {ChildStack}/{ResourceId}
+  
+  $ sam sync --code --watch --stack-name {stack}
+  
+  By default, the sync command runs a full AWS Cloudformation stack update. You can specify
+  --code or --watch to switch modes. Sync also supports nested stacks and
+  nested stack resources.
+
+  Running --watch with --code option will provide a way to run code
+  synchronization only, that will speed up start time and will skip any
+  template change. Please remember to update your deployed stack by running
+  without --code option.
+  
+  **This command requires access to AWS credentials.**
+
+Acronyms:
+
+    IAM: Identity and Access Management.
+    ARN: Amazon Resource Name.
+    SNS: Simple Notification Service.
+    ECR: Elastic Container Registy
+    KMS: Key Management Service.
+
+Options:
+
+  Required Options:
+
+  -t, --template-file, --template PATH    AWS SAM template file.  [default: template.[yaml|yml|json]]
+  --stack-name TEXT                       Name of the AWS CloudFormation stack. 
+
+  AWS Credential Options:
+  
+  Exploratory DESIGN NOTE: How can we streamline credentials
+  
+  --profile TEXT                  Named profile for AWS credentials.
+  --region TEXT                   Set the AWS Region. (e.g. us-east-1). 
+
+
+  Infrastructure Options:
+
+  --parameter-overrides           String that contains AWS CloudFormation parameter overrides encoded as
+                                  key=value pairs.
+                                  
+                                  For example, 'ParameterKey=KeyPairName,ParameterValue=MyKey
+                                  ParameterKey=InstanceType,ParameterValue=t1.micro' or KeyPairName=MyKey
+                                  InstanceType=t1.micro
+
+  --capabilities LIST             List of capabilities that one must specify before AWS Cloudformation can create
+                                  certain stacks.
+                                  More info at: https://docs.aws.amazon.com/serverlessrepo/latest/devguide/acknowledging-application-capabilities.html
```

Comment:
Would it be worth it to add the list of valid capabilities here so that we can save a click for those that are aware of what this option does?

Reply:
It was an exercise to just reduce the amount of text this had, adding the valid ones makes sense.